packages/babel-plugin-transform-inline-consecutive-adds/src/object-collapser.js

```diff
@@ -0,0 +1,51 @@
+"use strict";
+
+const Collapser = require("./collapser");
+
+class ObjectCollapser extends Collapser {
+  isInitTypeValid(init) {
+    return init.isObjectExpression();
+  }
+
+  isExpressionTypeValid(expr) {
+    return expr.isAssignmentExpression();
+  }
+
+  getExpressionChecker(objName, checkReference) {
+    return (expr) => {
+      // checks expr is of form:
+      // foo.a = rval
+
+      const left = expr.get("left");
+      if (!left.isMemberExpression()) {
+        return false;
+      }
+
+      const obj = left.get("object"), prop = left.get("property");
+      if (!obj.isIdentifier() || obj.node.name !== objName) {
+        return false;
+      }
+      if (!prop.isIdentifier() && checkReference(prop)) {
```

Comment:
Ah. I should've added this in the place where you add it to the list of properties